### PR TITLE
[FineObjLite] Refactor headers

### PR DESCRIPTION
**Pull request by favorart**

### CODE REVIEW

File: NeoML/include/NeoML/FineObjLite/ArrayFOL.h [hunk 1]
```diff
@@ -92,9 +95,11 @@ class CArray {
 
 	CArray();
 	CArray( std::initializer_list<T> list );
+	CArray( CArray&& ) = default;
```
Comment (AndrewAndrianov) [inline NeoML/include/NeoML/FineObjLite/ArrayFOL.h]: Is this really possible?
see [Array.h](https://github.com/abbyyProduct/Tools_Libs-FineObjects15/blob/aa7fa3de161442fc466db39aeb9049fe5c3e1b9d/Inc/Array.h#L349)

File: NeoML/include/NeoML/FineObjLite/FastArrayFOL.h [hunk 2]
```diff
@@ -31,9 +31,11 @@ class CFastArray {
 
 	CFastArray();
 	CFastArray( std::initializer_list<T> list );
+	CFastArray( CFastArray&& ) = default;
```
Comment (AndrewAndrianov) [inline NeoML/include/NeoML/FineObjLite/FastArrayFOL.h]: This is also doubtful: [FastArray.h](https://github.com/abbyyProduct/Tools_Libs-FineObjects15/blob/aa7fa3de161442fc466db39aeb9049fe5c3e1b9d/Inc/FastArray.h#L305)

File: cmake/Utils.cmake [hunk 3]
```diff
@@ -93,7 +93,7 @@ macro(add_gtest_target)
         FetchContent_Declare(
             GoogleTest
             GIT_REPOSITORY https://github.com/google/googletest.git
-            GIT_TAG release-1.10.0
+            GIT_TAG release-1.11.0
```
Comment (AndrewAndrianov) [inline cmake/Utils.cmake]: This edit seems to be unnecessary here

File: NeoML/include/NeoML/FineObjLite/ArrayFOL.h (hunk 1 above)
Comment (Eugene-Alexeev) [reply NeoML/include/NeoML/FineObjLite/ArrayFOL.h]: default `move` will call move by members, if someone can be moved bitwise but this is not specified in `IsMemmoveable`, then it’s extra work

File: NeoML/include/NeoML/FineObjLite/PointerArrayFOL.h [hunk 4]
```diff
@@ -18,24 +18,28 @@ limitations under the License.
...
 template<class T, class Allocator = CurrentMemoryManager>
-class CPointerArray {
+class CPointerArray final {
 public:
 	typedef T* TElement;
 	typedef Allocator AllocatorType;
 
-	CPointerArray();
-	~CPointerArray();
+	CPointerArray() {}
```
Comment (Eugene-Alexeev) [inline NeoML/include/NeoML/FineObjLite/PointerArrayFOL.h]: `default` can also be done

File: NeoML/include/NeoML/FineObjLite/PriorityQueueFOL.h [hunk 5]
```diff
@@ -18,14 +18,17 @@ limitations under the License.
...
-class CPriorityQueue {
+class CPriorityQueue final {
 public:
 	typedef ARRAY ArrayType;
 	typedef COMPARE CompareType;
 	typedef CPriorityQueue<ArrayType, CompareType> QueueType;
 	typedef typename ARRAY::TElement ElementType;
 
 	explicit CPriorityQueue( const CompareType& compare = CompareType() ) : data( compare ) {}
+	CPriorityQueue( CPriorityQueue&& ) = default;
```
Comment (Eugene-Alexeev) [inline NeoML/include/NeoML/FineObjLite/PriorityQueueFOL.h:30]: in modern C++ it is also recommended to mark `move assign`/`move constructor` as `noexcept` where possible.
https://gieseanw.wordpress.com/2020/08/28/friendly-reminder-to-mark-your-move-constructors-noexcept/

File: cmake/Utils.cmake (hunk 3 above)
Comment (favorart) [reply cmake/Utils.cmake]: fixed

File: NeoML/include/NeoML/FineObjLite/PriorityQueueFOL.h (hunk 5 above)
Comment (pavel-voropaev) [reply NeoML/include/NeoML/FineObjLite/PriorityQueueFOL.h:30]: Still this class contains ArrayType with non-noexcept move-functions... And [here](https://github.com/abbyyProduct/Tools_Libs-FineObjects15/pull/158#discussion_r1620520076) we were close to eliminate _noexcept_ from everywhere.

File: NeoML/include/NeoML/FineObjLite/PointerArrayFOL.h (hunk 4 above)
Comment (favorart) [reply NeoML/include/NeoML/FineObjLite/PointerArrayFOL.h]: fixed

File: NeoML/include/NeoML/FineObjLite/PriorityQueueFOL.h (hunk 5 above)
Comment (favorart) [reply NeoML/include/NeoML/FineObjLite/PriorityQueueFOL.h:30]: fixed

File: NeoML/include/NeoML/FineObjLite/ArrayFOL.h (hunk 1 above)
Comment (favorart) [reply NeoML/include/NeoML/FineObjLite/ArrayFOL.h]: fixed

File: NeoML/include/NeoML/FineObjLite/FastArrayFOL.h (hunk 2 above)
Comment (favorart) [reply NeoML/include/NeoML/FineObjLite/FastArrayFOL.h]: fixed

**Review by AndrewAndrianov — APPROVED** [review]

**Pull request merged**